chore: switch images of llama.cpp to the RamaLama images

benoitf opened the pull request:
### What does this PR do?
switch to the RamaLama images

Use default images for macOS with gpuLayers=999 (as RamaLama does for libkrun/macOS) rather than using vulkan images
(vulkan images are failing with libKrun (unstable on my end))

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop-extension-ai-lab/issues/2630

### How to test this PR?

try to start a playground/service for a model

axel7083 commented on packages/backend/src/workers/provider/LlamaCppPython.ts:
```diff
@@ -173,7 +173,13 @@ export class LlamaCppPython extends InferenceProvider {
+
+        // set the number of layers to offload to the GPU in case of LibKrun
+        if (vmType === VMType.LIBKRUN || vmType === VMType.LIBKRUN_LABEL) {
+          envs.push(`GPU_LAYERS=999`);
```
question (non-blocking): `-1` does not work as max?

benoitf replied:
no unfortunately it doesn't

at the beginning I had errors as the 'cpu part' was used and not the GPU

comparing the arguments to start llama-server and AI Lab lead me to this difference


in RamaLama docs there is:

**ngl**=-1
number of gpu layers, 0 means CPU inferencing, 999 means use max layers (default: -1)
The default -1, means use whatever is automatically deemed appropriate (0 or 999)

(so it seems it's picking up 0 :-/ )

benoitf replied:
and https://github.com/containers/ramalama/pull/802

benoitf commented:
note: I tested on macOS with a libkrun podman machine, I haven't tested on Windows

axel7083 reviewed (approved):
## Linux native

- Documentation is accessible
- Playground is working as expected
- `/v1/chat/completion` works well from PostMan

jeffmaury commented:
Model refuses to start: MaziyarPanahi/Mistral-7B-Instruct-v0.3.Q4_K_M
```

warning: no usable GPU found, --gpu-layers option will be ignored
warning: one possible reason is that llama.cpp was compiled without GPU support
warning: consult docs/build.md for compilation instructions
terminate called after throwing an instance of 'std::runtime_error'
  what():  error: the supplied chat template is not supported: chatml-function-calling
note: llama.cpp was started without --jinja, we only support commonly used templates

�/usr/bin/llama-server.sh: line 14:     2 Aborted                 (core dumped) llama-server --model ${MODEL_PATH} --host ${HOST:=0.0.0.0} --port ${PORT:=8001} --gpu_layers ${GPU_LAYERS:=0} ${CHAT_FORMAT}
```

jeffmaury commented:
Also I am not getting model metrics with bartowski/granite-3.1-8b-instruct-GGUF works with TheBloke/Mistral-7B-Instruct-v0.2-GGUF

benoitf commented:
for the crash it's that the chat template is not available


current list is chatglm3, chatglm4, chatml, command-r, deepseek, deepseek2, exaone3, gemma, granite, llama2, llama2-sys, llama2-sys-bos, llama2-sys-strip, llama3, minicpm, mistral-v1, mistral-v3, mistral-v3-tekken, mistral-v7, monarch, openchat, orion, phi3, rwkv-world, vicuna, vicuna-orca, zephyr

wanted `chatml-function-calling` which is not there

benoitf commented:
template for function calling updated in latest commit to `chatml`

gastoner reviewed (approved):
LGTM on Windows without a GPU

benoitf commented:
rebased and temporary images for now (until the next release of RamaLama is out) just to try the PR

it includes this change from RamaLama https://github.com/containers/ramalama/pull/1053

I'm able to run the functional programming recipe

axel7083 reviewed (changes requested):
On windows running model `MaziyarPanahi/Mistral-7B-Instruct-v0.3.Q4_K_M` the server is not starting

Logs

```
{"msg":"exec container process `/usr/bin/llama-server.sh`: Exec format error","level":"error","time":"2025-03-28T14:30:48.439048Z"}
```

benoitf commented:
@axel7083 I think this is expected for now as my image is arm64 only. I'm waiting for official image ( next release)

axel7083 commented:
> @axel7083 I think this is expected for now as my image is arm64 only. I'm waiting for official image ( next release)

Good to know, tag me when I will need to test again 👍

benoitf commented:
@axel7083 0.7.2 images of RamaLama are now available since few minutes, so I switched to these images

axel7083 reviewed (commented):
The CPU image is working, however the cuda image does not start and get the following error

```
chmod: cannot access './run.sh': No such file or directory
```

> Sorry it took so long to review, the image is 6.75 GB and my internet is not nice

benoitf commented:
maybe then https://github.com/containers/ramalama/pull/1081/files

benoitf commented:
forgot to notify here that I switched to the respin of the images yesterday evening

ScrewTSW commented:
![Screenshot_20250401_112510](https://github.com/user-attachments/assets/b6672c74-618f-44c2-a0ab-4ff96ccf8c3d)
currently cannot start projects on GPU, Linux, tgz, Nvidia RTX 3090

benoitf commented:
I updated the images of RamaLama

I also update the entrypoint to be used and removed the chmod operation as the script has correct permissions

it should now work on Windows and Linux

axel7083 commented on packages/backend/src/assets/inference-images.json:
```diff
@@ -3,8 +3,7 @@
-    "cuda": "ghcr.io/containers/podman-desktop-extension-ai-lab-playground-images/ai-lab-playground-chat-cuda@sha256:e4b57e52c31b379b4a73f8e9536bc130fdea665d88dbd05643350295b3402a2f",
-    "vulkan": "ghcr.io/containers/podman-desktop-extension-ai-lab-playground-images/ai-lab-playground-chat-vulkan@sha256:6a93b247099643f4f8c78ee9896c2ce4e9a455af114a69be09c16ad36aa51fd2"
+    "default": "quay.io/ramalama/ramalama-llama-server@sha256:cbadb36fbbc2abf9867a33e6dfe3f2df4a76774259b5d4d24d50f4fc7e525406",
+    "cuda": "quay.io/ramalama/cuda-llama-server@sha256:cbadb36fbbc2abf9867a33e6dfe3f2df4a76774259b5d4d24d50f4fc7e525406"
```
Can't pull this image?

````
> podman pull quay.io/ramalama/cuda-llama-server@sha256:cbadb36fbbc2abf9867a33e6dfe3f2df4a76774259b5d4d24d50f4fc7e525406
Trying to pull quay.io/ramalama/cuda-llama-server@sha256:cbadb36fbbc2abf9867a33e6dfe3f2df4a76774259b5d4d24d50f4fc7e525406...
Error: initializing source docker://quay.io/ramalama/cuda-llama-server@sha256:cbadb36fbbc2abf9867a33e6dfe3f2df4a76774259b5d4d24d50f4fc7e525406: reading manifest sha256:cbadb36fbbc2abf9867a33e6dfe3f2df4a76774259b5d4d24d50f4fc7e525406 in quay.io/ramalama/cuda-llama-server: manifest unknown
````

![image](https://github.com/user-attachments/assets/e29e6ebf-801c-494a-bd23-17323f144014)

benoitf replied:
🤦 the tag has been overriden so the sha is no longer there, updating...

axel7083 reviewed (commented):
Seems that the cuda image is 404 ?

axel7083 commented:
> ok so basically I had to apply the env for libkrun but we need that for Windows/NVidia as well

yes, before it was working as -1 was max out, but now -1 give 0 offloaded

jeffmaury reviewed (approved):
LGTM.

tested some recipes and models (ok not all combinations) without issues on Windows

ScrewTSW reviewed (approved):
LGTM, ramalama inference starts
